core/main.go

```diff
@@ -125,6 +128,62 @@ func (c *Core) ConnectToDatabase() {
 	log.Debugf("connected successfully to database!")
 }
 
+func (c *Core) InitializePrometheus() {
+	tenants, err := c.db.GetAllTenants(nil)
+	if err != nil {
+		log.Debugf("No tenants found in database")
```

Comment:
The error doesn't mean that there are no tenants. It means that there was a problem fetching the tenants.

Reply:
Use `MaybeTerminate` to bail if this goes wrong.

Reply:
Do this for all the instances in this function.